dev-util/serve-d/serve-d-0.8.0_beta15-r2.ebuild

```diff
@@ -0,0 +1,213 @@
+https://code.dlang.org/packages/sdlfmt/0.1.1.zip -> sdlfmt-0.1.1.zip
+https://code.dlang.org/packages/sdlite/1.1.2.zip -> sdlite-1.1.2.zip
+https://code.dlang.org/packages/silly/1.1.1.zip -> silly-1.1.1.zip
+https://code.dlang.org/packages/standardpaths/0.8.2.zip -> standardpaths-0.8.2.zip
+https://code.dlang.org/packages/stdx-allocator/2.77.5.zip -> stdx-allocator-2.77.5.zip
+https://code.dlang.org/packages/taggedalgebraic/0.11.22.zip -> taggedalgebraic-0.11.22.zip
+https://code.dlang.org/packages/test_allocator/0.3.4.zip -> test_allocator-0.3.4.zip
+https://code.dlang.org/packages/unit-threaded/0.10.8.zip -> unit-threaded-0.10.8.zip
+https://code.dlang.org/packages/vibe-core/2.2.0.zip -> vibe-core-2.2.0.zip
+https://code.dlang.org/packages/vibe-d/0.9.6.zip -> vibe-d-0.9.6.zip
+https://code.dlang.org/packages/xdgpaths/0.2.5.zip -> xdgpaths-0.2.5.zip
+"
+
+LICENSE="MIT"
+LICENSE+=" Apache-2.0 BSD-2 BSD Boost-1.0 GPL-3 ISC LGPL-3 MIT public-domain Unlicense || ( openssl SSLeay )"
+S="${WORKDIR}/${PN}-${MY_VER}"
+SLOT="0"
+KEYWORDS="~amd64"
+
+# gdc currently fails due to a bug in mir-cpuid, see: https://github.com/libmir/mir-cpuid/pull/46
+DLANG_COMPAT=( dmd-2_{106..107} ldc2-1_{35..36} )
+
+CHECKREQS_MEMORY="10G" # mir is a chonker
+
+inherit check-reqs dlang-single multiprocessing
+
+DEPEND=${DLANG_DEPS}
+# Lower versions of dcd won't immediately fail but they won't work as
+# intended (no autocompletion for example).
+RDEPEND=">=dev-util/dcd-0.15.2 ${DEPEND}"
+BDEPEND="dev-util/dub app-arch/unzip ${DEPEND}"
+REQUIRED_USE=${DLANG_REQUIRED_USE}
+
+src_unpack() {
+	unpack "${P}.zip"
+	pushd "${S}" || die
+
+	local dep name ver dub_args
+	dub_args=(
+		# Don't look up dependencies online or in the system path
+		--skip-registry=all
+		# Only look for them in the ${T} directory
+		--registry=file://"${T}"
+		# Prefer verbose operation
+		--verbose
+		# Unpack dependencies for the local project (put them in ${S}/.dub)
+		--cache=local
+	)
+	for dep in ${A}; do
+		if [[ ${dep} != ${P}.zip ]]; then
+			# Due to a bug in dub, we can't have similar looking
+			# archives in the same directory. Simply put, if we had:
+			# foo-1.zip and foo-bar-2.zip when trying to fetch foo dub
+			# will fail because it thinks that foo-bar-2.zip is foo's
+			# archive. See: https://github.com/dlang/dub/pull/2727
+			cp "${DISTDIR}/${dep}" "${T}" || die "Could not copy dependency to temporary directory"
+
+			# This calculation is faulty, dub allows digits in package names, it will
+			# probably be needed to save dependencies in a similar to $CRATES variable.
+			name="${dep%%-[0-9]*}"
+			version="${dep%.zip}"
+			version="${version#${name}-}"
+
+			"${EPREFIX}"/usr/bin/dub "${dub_args[@]}" fetch "${name}@${version}" \
+				|| die "Could not extract dub dependencies"
+
+			rm "${T}/${dep}" || die "Could not remove depedency from temporary directory"
+		fi
+	done
+	popd > /dev/null || die
+}
+
+src_compile() {
+	local dub_args=(
```

Comment:
Feel like this belongs in an eclass too (dub_args overall). Add an eclass helper for calling dub build?  edub?

Reply:
Right, dub.eclass has been a thought that occurred to me and I didn't dismiss it entirely. The problem with it is mostly my perfectionist nature, which makes me want to have a set of functionalities, like the edub helper you mentioned, a way to specify dependencies and have them be auto-extracted/auto-added(there is a way to tell dub to use a local copy of packages which is what I've seen be used by cargo, I couldn't figure it out on my first try so that's why the ebuild currently lets dub unpack stuff), a script that figures out LICENSE based on a dub project and ideally transforms the license names to what is used by Gentoo, and, if feeling fancy, some custom test runners that handle popular test-suites, the features just don't end.

If I do this, however, I will need some more packages to test stuff to make sure the eclass does its job as an eclass. Adding more packages takes time, as well as learning how to implement functionalities based on current stuff in ::gentoo, so that's the main issue why I have been postponing working on it. I want first to get stuff fixed (like adding tests to dev-lang/{dmd,ldc2}, making sure user flags are respected, make sure $EPREFIX works) before I start adding new features.

Now, these have all been arguments for why I'm not working on a *fully*-featured dub.eclass, and they don't necessarily go against your request. I believe that on stuff that I'm uncertain I should just go with your opinion, that probably has the best chances of turning right in the end. So, I can implement a small dub.eclass with only an edub funtion and probably some support for auto-unpacking, that should be simple enough or leave the ebuild how it is for a while and come back to dub.eclass when I can give it more attention and have a more proper solution, which one do you think I should do?